src/Microsoft.AspNet.Authentication/Events/SigningInContext.cs

```diff
@@ -25,6 +28,13 @@ public abstract class ReturnEndpointContext : EndpointContext
         public ClaimsPrincipal Principal { get; set; }
         public AuthenticationProperties Properties { get; set; }
 
+        public bool IsRequestCompleted { get; private set; }
+
+        public void RequestCompleted()
+        {
+            IsRequestCompleted = true;
```

Comment:
Set this inline on the property, and remove the `private set;` (it's implied in C# 6).